From 38ca57a9cd5d6555f2c1981cd3974203350f2dbc Mon Sep 17 00:00:00 2001
From: James Ausmus <james.ausmus@intel.com>
Date: Thu, 10 Oct 2013 19:05:12 -0700
Subject: Re-create dirty merge 32c913e

Fix code up to post-merge state after 32c913e4369ce7bd1d16a9b6983f7b8975c13f5a
("Merge tag 'drm-intel-next-2013-07-26-fixed' of
git://people.freedesktop.org/~danvet/drm-intel into drm-next")

Signed-off-by: James Ausmus <james.ausmus@intel.com>
(cherry picked from
 https://chromium.googlesource.com/chromiumos/third_party/kernel-next
 chromeos-3.10, 3a0d8e0095892f31c9cdde50b147ea606bb541b7)
Signed-off-by: Darren Hart <dvhart@linux.intel.com>
---
 drivers/gpu/drm/i915/i915_gem_stolen.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_stolen.c b/drivers/gpu/drm/i915/i915_gem_stolen.c
index 0d0a3b179075..ba4dabca83c2 100644
--- a/drivers/gpu/drm/i915/i915_gem_stolen.c
+++ b/drivers/gpu/drm/i915/i915_gem_stolen.c
@@ -423,7 +423,8 @@ i915_gem_object_create_stolen_for_preallocated(struct drm_device *dev,
 	return obj;
 
 err_out:
-	drm_mm_put_block(stolen);
+	drm_mm_remove_node(stolen);
+	kfree(stolen);
 	drm_gem_object_unreference(&obj->base);
 	return NULL;
 }
-- 
1.8.5.rc3

